docs/rst/tutorials/aml_pipeline.rst

```diff
@@ -1,24 +1,452 @@
+
+
+.. code-block:: python
+
+    from MNIST import mnistGenerator
+
+The ``mnistGenerator`` searches and loads the datasets and provides functionality to load individual digits.
+
+Embedding constants
+===================
+
+A simple embedding for black and white images:
+- constants for each black pixel `B(i,j)` (height x width),
+- constants for each white pixel `W(i,j)` (height x width),
+- constants for each label (10 if using all digits).
+
+For instance, with 2x2 images the embedding set would be
+``[B(1,1), B(1,2), B(2,1), B(2,2), W(1,1), W(1,2), W(2,1), W(2,2), label(1), label(2), ...]``
+
+### Embedding function
+
+We can use the following function to embed MNIST images.
+
+An image is defined by a set of constants, one per pixel. Each constant will describe that pixel as white or black.
+For instance:
+``[W(1,1), W(1,2), B(1,3), B(1,4), W(1,5), ... W(28,27), B(28,28)]``
+
+
+.. code-block:: python
+
+    def digitToConstants(d, size):
+        ss = size * size
+        printImage = False
+
+        j = 0
+        threshold = 256 / 10
+        ret = [-1] * size * size
+        for x in range(size):
+            for y in range(size):
+                if d[j] > threshold:
+                    ret[j] = j
+                else:
+                    ret[j] = j + ss
+                j += 1
+
+        result = set(ret)
+        return result
+
+Functions to read examples and embed them as constants
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+These functions load images and embed them.
+
+- ``mnistDigit`` produces embedded images from the target class, to create positive duples.
+- ``mnistOtherDigit`` produces images from any class other than the target, to create negative duples.
+
+
+.. code-block:: python
+
+    def mnistDigit(targetDigit, size):
+        d, label, i = MNIST_GENERATOR.getNextDigit(targetDigit, False)
+        result = digitToConstants(d, size)
+        return sc.amlset(result)
+
+
+    def mnistOtherDigit(targetDigit, size):
+        d, label, i = MNIST_GENERATOR.getNextDigit(targetDigit, True)
+        result = digitToConstants(d, size)
+        return sc.amlset(result)
+
+Training
+========
+
+Training parameters
+^^^^^^^^^^^^^^^^^^^
+
+
+.. code-block:: python
+
+    # Training iterations
+    iterations = 1000 # Maximum number of mini-batch iterations
+
+    # Problem parameters
+    validation_size = 10000
+    MNIST_GENERATOR = mnistGenerator(validation_size)
+    gridSideLength = 28  # side of images in the MNIST dataset
+    embedding_constants = {*range(0, 2 * gridSideLength * gridSideLength)}
+    targetDigit = 0  # 0 ... 9
+
+    balance = 9
+    pBatchSize = 100
+    nBatchSize = 100 * balance
+    maxPTrainingExamples = 6000
+    maxNTrainingExamples = 54000
+
+Notice that we are picking one target digit.
+The model will discriminate between that digit (positive class), and the rest.
+It is also possible to slightly modify this script for the multiclass problem.
```

Comment:
Move this sentence to another section at the end of this tutorial called "Next Steps" and add a link there to another tutorial named "Extending the MNIST binary classifier to a multiclass classifier". Create the file for this tutorial and add it to the index. Then on the file just add the title and a warning with a "Work in progress" statement.

Reply:
No sé si es una buena idea. Es sencillo extender el código a multiclase, pero la inferencia se vuelve mucho más complicada (a nivel estadístico, no de AML)

Reply:
Let's skip it for now then